### PR TITLE
Feature/platform interface

### PR DESCRIPTION
**Pull request by diptorupd**

### CODE REVIEW

File: dpctl-capi/source/dpctl_sycl_platform_interface.cpp [hunk 1]
```diff
@@ -37,152 +37,187 @@ using namespace cl::sycl;
...
+    try {
+        auto P = new platform();
+        PRef = wrap(P);
+    } catch (std::bad_alloc const &ba) {
+        std::cerr << ba.what() << '\n';
+    }
+    return PRef;
+}
+
+__dpctl_give DPCTLSyclPlatformRef DPCTLPlatform_CreateFromSelector(
```
Comment (oleksandr-pavlyk) [inline dpctl-capi/source/dpctl_sycl_platform_interface.cpp:76]: Cool. I was not aware platform could be created from a device selector.

File: dpctl/_sycl_platform.pyx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        DPCTLCString_Delete(self._name)
+        DPCTLCString_Delete(self._vendor)
+        DPCTLCString_Delete(self._version)
+
+
+cdef class SyclPlatform(_SyclPlatform):
+    """ Python equivalent for cl::sycl::platform class.
+    """
+    @staticmethod
+    cdef void _init_helper(SyclPlatform platform, DPCTLSyclPlatformRef PRef):
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_platform.pyx:70]: Arg should be `_SyclPlatform` type.

File: dpctl/_sycl_platform.pyx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    def __repr__(self):
+        return ("<dpctl." + self.__name__ + " [" +
+                self.name + ", " + self.vendor +", " +
+                " " + self.version + "] at {}>".format(hex(id(self))) )
+
+    def __cinit__(self, arg=None):
+        if type(arg) is unicode:
+            string = bytes(<unicode>arg, "utf-8")
+            filter_c_str = string
+            DSRef = DPCTLFilterSelector_Create(filter_c_str)
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_platform.pyx:118]: This branch and the next one share code, that can be factored out into a separate `_init_from_cstring` method

File: dpctl/_sycl_platform.pyx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    about each platform.
+
+    Currently, this function prints a list of all SYCL platforms that
+    are available on the system and the list of devices for each platform.
+
+    :Example:
+        On a system with an OpenCL CPU driver, OpenCL GPU driver,
+        Level Zero GPU driver, running the command. ::
+
+        $python -c "import dpctl; dpctl.print_info()"
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_platform.pyx:214]: Should this docstring reflect the actual name of the function, i.e. replace `dpctl.print_info()` with `dpctl.lsplatform()` ?

File: dpctl/_sycl_platform.pyx (hunk 2 above)
Comment (diptorupd) [reply dpctl/_sycl_platform.pyx]: done.

File: dpctl/_sycl_platform.pyx (hunk 3 above)
Comment (diptorupd) [reply dpctl/_sycl_platform.pyx]: done.

File: dpctl/_sycl_platform.pyx (hunk 4 above)
Comment (diptorupd) [reply dpctl/_sycl_platform.pyx]: done.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Pull request merged**